pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java

```diff
@@ -813,4 +813,21 @@
      * @param enabled whether to enable AutoScaledReceiverQueueSize.
      */
     ConsumerBuilder<T> autoScaledReceiverQueueSizeEnabled(boolean enabled);
+
+    /**
+     * Configure topic specific options to override those set at the {@link ConsumerBuilder} level.
```

Comment:
Question:
Do we have the priority for topic name or pattern?
If the user registers two configurations, one is the full topic name and another is the pattern. which one is better to choose?
E.g:
1. "test-topic-name"
2. "test-*"

Reply:
In the PIP issue (#16481) I wrote:

> Since the topic configuration is effectively keyed by a topic name or pattern, it’s possible for the user to be able configure multiple topic configurations that match the same concrete topic name. In this case, the first topic name match should be selected.

So the first matching topic configuration that was registered will be used. I'm happy to consider alternative approaches.

Reply:
Got it, thanks!